site/source/tutorials/working_with_virtual_dom.md

```diff
@@ -0,0 +1,106 @@
+
+## Advantages of virtualizing the DOM
+
+> "We can solve any problem by introducing an extra level of indirection" - David J. Wheeler
+
+Ensuring that the DOM is updated efficiently is a critical component to maximizing a web application's performance. Doing that consistently across a large application, however, can be very challenging. Rather than relying on developers to manage this manually, Dojo 2 uses an abstraction layer to manage DOM updates automatically. This abstraction layer is called a *virtual* DOM.
+
+A virtual DOM has several advantages when compared to traditional DOM manipulation. A virtual DOM improves developer productivity by allowing them to interact with a simple, consistent API in a declarative manner. It also improves performance by gathering DOM updates into batches and applying them at the optimum time. Finally, applications using a virtual DOM use resources more efficiently since applications interact with lightweight representations of DOM elements instead of the DOM elements themselves.
+
+The DOM API has existed for many years and has been continually expanded to keep up with the evolution of web browsers. This evolution has yielded a programming interface that is occasionally verbose and inconsistent. Virtual DOM libraries are able to reconsider the use-cases for managing page elements and expose a simpler set of interfaces for developers to work with. For example, consider the example in the preceeding section where we created an `h1` tag, set some properties on it, and add it to the page. In Dojo 2, this would be accomplished using the following code:
+
+```ts
+  v('h1', { class: 'title' }, [ 'Welcome to Biz-E-Corp' ]);
+```
+
+A single line of code is all that is required to accomplish what the DOM API required four to do. By providing a simple, declarative API, rendering code is typically much cleaner and easier to understand.
+
+One critical difference between the DOM and virtual DOM examples is that the DOM example maintains access to the element that was created via the `pageTitle` variable. Keeping a reference like this is often done when the element needs to be changed later. With a virtual DOM, maintaining state in this way is not necessary because the application never manipulates the state of the DOM directly. It is only responsible for generating a representation of the page at any given time. A component of the virtual DOM called a [projector](http://maquettejs.org/docs/typedoc/interfaces/_maquette_.projector.html) is responsible to determine the changes that are required to the page.
```

Comment:
Keeping a reference like this is often done when the element needs to be changed later -> Retaining this reference is often done when an element may need to be modified later